features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/CheckableTextViewMapper.kt

```diff
@@ -109,6 +111,11 @@ internal abstract class CheckableTextViewMapper<T>(
                 view,
                 mappingContext.systemInformation.screenDensity
             )
+            val drawableCopier = object : DrawableCopier {
```

Comment:
you can add `fun` to the definition of `DrawableCopier` making it a functional interface: `fun interface DrawableCopier`.

This will allow you to write simply `val drawableCopier = DrawableCopier { originalDrawable, resources -> cloneCheckableDrawable(view, it) }`.

But anyway, since it is a stateless object maybe it is not worth to allocate a new one on each call, so you can lift it to the class property.

Reply:
Updated with `fun interface`, meanwhile it's not stateless object, for example this implementation relies on the view instance passed from the function.

Reply:
Yes, but there is no any property in this object, so all the changes are local to the function and don't leave that scope.

Reply:
Yes I agree that there is no property for the object, but it still seems not possible to lift it to the class property due to the usage of view instance.